File: scripts/python/data_generation.py
import pandas as pd
import numpy as np
import os

np.random.seed(42)
n = 1000

df = pd.DataFrame({
    "product_name": ["Trulicity"] * n,
    "region": np.random.choice(["North America", "Europe", "Asia"], size=n),
    "regulatory_body": np.random.choice(["FDA", "EMA", "PMDA"], size=n),
    "clinical_trial_phase": np.random.choice(["I", "II", "III", "IV"], size=n),
    "adverse_event_reports": np.random.poisson(5, size=n),
    "manufacturing_site": np.random.choice(["SiteA", "SiteB", "SiteC"], size=n),
    "inspection_findings": np.random.poisson(2, size=n),
    "compliance_training": np.random.binomial(1, 0.7, size=n),
    "previous_violations": np.random.binomial(3, 0.1, size=n),
    "inspection_report_text": np.random.choice([
        "No critical issues noted.", "Documentation errors noted.",
        "Data integrity concerns.", "Untrained personnel.",
        "No findings.", "Deviation found."
    ], size=n),
    "compliance_notes_text": np.random.choice([
        "All procedures followed.", "Deviation due to site.",
        "Docs not up to date.", "Training records missing.",
        "QA pending review."
    ], size=n)
})

logit = (
    -1 + 0.3 * df["adverse_event_reports"] +
    0.5 * df["inspection_findings"] +
    0.8 * df["previous_violations"] -
    0.6 * df["compliance_training"]
)
prob = 1 / (1 + np.exp(-logit))
df["compliance_issue"] = np.random.binomial(1, prob)

os.makedirs("outputs", exist_ok=True)
df.to_csv("outputs/synthetic_compliance_data.csv", index=False)

File: scripts/r/data_generation.R
#### In a non-notional scenario, this script would be replaced by executing SQL against company databases of record, RAG-enabled gathering, or manual methods ####
#
# Typically I handle data ingest and cleaning in separate scripts but this would be subject to company preferred practice and code efficiency.

set.seed(42)
n <- 1000

product_name <- rep("Trulicity", n)
region <- sample(c("North America", "Europe", "Asia"), n, replace = TRUE)
regulatory_body <- sample(c("FDA", "EMA", "PMDA"), n, replace = TRUE)
clinical_trial_phase <- sample(c("I", "II", "III", "IV"), n, replace = TRUE)
adverse_event_reports <- rpois(n, 5)
manufacturing_site <- sample(c("SiteA", "SiteB", "SiteC"), n, replace = TRUE)
inspection_findings <- rpois(n, 2)
compliance_training <- rbinom(n, 1, 0.7)
previous_violations <- rbinom(n, 3, 0.1)

inspection_texts <- c(
  "No critical issues noted.",
  "Documentation errors noted.",
  "Data integrity concerns.",
  "Untrained personnel.",
  "No findings.",
  "Deviation found."
)

compliance_notes <- c(
  "All procedures followed.",
  "Deviation due to site.",
  "Docs not up to date.",
  "Training records missing.",
  "QA pending review."
)

inspection_report_text <- sample(inspection_texts, n, replace = TRUE)
compliance_notes_text <- sample(compliance_notes, n, replace = TRUE)

logit <- -1 + 0.3 * adverse_event_reports + 0.5 * inspection_findings +
  0.8 * previous_violations - 0.6 * compliance_training
prob <- 1 / (1 + exp(-logit))
compliance_issue <- rbinom(n, 1, prob)

df <- data.frame(
  product_name, region, regulatory_body, clinical_trial_phase,
  adverse_event_reports, manufacturing_site, inspection_findings,
  compliance_training, previous_violations,
  inspection_report_text, compliance_notes_text,
  compliance_issue
)

write.csv(df, "data/synthetic_compliance_data.csv", row.names = FALSE)

